cryptopan/src/cryptopan.rs

```diff
@@ -0,0 +1,225 @@
+//! `CryptoPAn`
+//! IP-address anonymization algorithm that preserves prefixes
+//! Tightly based on the GO implementation
+//! Author: Yawning Angel <yawning at schwanenlied dot me>
+//! [Github Link](./https://github.com/Yawning/cryptopan/blob/master/cryptopan.go)
+//! Based upon the algorithm described in:
+//! Package cryptopan implements the Crypto-PAn prefix-preserving IP address sanitization algorithm as specified by J. Fan, J. Xu, M. Ammar, and S. Moon.
+use aes::cipher::BlockEncryptMut;
+use aes::cipher::{generic_array::GenericArray, KeyInit};
+use bitvec::field::BitField;
+use bitvec::prelude::Msb0;
+use bitvec::{bitvec, vec::BitVec};
+use std::net::{
+    IpAddr,
+    IpAddr::{V4, V6},
+    Ipv4Addr, Ipv6Addr,
+};
+
+type Aes128EcbEnc = ecb::Encryptor<aes::Aes128>;
+
+pub struct CryptoPAn {
```

Comment:
Maybe make this type generic over used ciphers with AES-128 being the default one? Also add some docs to the type.